tools/rosbag/src/rosbag/migration.py

```diff
@@ -271,23 +274,26 @@ def __init__(self, migrator, location):
         # Instantiate types dynamically based on definition
         try:
             if self.old_type == "":
-                raise Exception
+                raise self.EmptyType
             self.old_types = genpy.dynamic.generate_dynamic(self.old_type, self.old_full_text)
             self.old_class = self.old_types[self.old_type]
             self.old_md5sum = self.old_class._md5sum
-        except:
-            self.old_types = []
+        except Exception as e:
+            if not isinstance(e, self.EmptyType):
+                print("Warning: {}".format(e))
```

Comment:
Printing a full traceback might be more helpful (especially if it is unclear when this might happen). Also warnings / errors should go to `stderr`.

Reply:
Sure for `stderr`.

Regarding the traceback, I can print it, yes. I was not sure because the previous behavior was to just silently ignore errors. I added the warning to be as close as possible to the previous behavior but I do not have example of when it happens. Maybe it would just be better to only catch the `EmptyType` exception instead.

Reply:
> Maybe it would just be better to only catch the `EmptyType` exception instead.

That would change / break behavior which would be problematic to be shipped into an existing ROS distro. I think a message with a traceback would be acceptable in terms of noise (with the advantage of actually knowing what failed).